modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/internal/search/spi/model/index/contributor/CommerceCurrencyModelDocumentContributor.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.currency.internal.search.spi.model.index.contributor;
+
+import com.liferay.commerce.currency.model.CommerceCurrency;
+import com.liferay.portal.kernel.search.Document;
+import com.liferay.portal.kernel.search.Field;
+import com.liferay.portal.kernel.util.Localization;
+import com.liferay.portal.search.spi.model.index.contributor.ModelDocumentContributor;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Mahmoud Azzam
+ */
+@Component(
+	property = "indexer.class.name=com.liferay.commerce.currency.model.CommerceCurrency",
+	service = ModelDocumentContributor.class
+)
+public class CommerceCurrencyModelDocumentContributor
+	implements ModelDocumentContributor<CommerceCurrency> {
+
+	@Override
+	public void contribute(
+		Document document, CommerceCurrency commerceCurrency) {
+
+		String commerceCurrencyDefaultLanguageId =
+			_localization.getDefaultLanguageId(commerceCurrency.getName());
+
+		String[] languageIds = _localization.getAvailableLanguageIds(
+			commerceCurrency.getName());
+
+		for (String languageId : languageIds) {
+			String code = commerceCurrency.getCode();
+
+			String name = commerceCurrency.getName(languageId);
+
+			document.addText(Field.CONTENT, name);
+
+			document.addText(
+				_localization.getLocalizedName(Field.NAME, languageId), name);
+
+			document.addText(Field.CODE, code);
```

Comment:
I don't think it's necessary to index the code for every language, it should be language independent, isn't it?

Reply:
true, code is not a localized field